src/index.rs

```diff
@@ -83,29 +82,62 @@ impl ResourceIndex {
             root: root_path.clone(),
         };
 
+        // We should not return early in case of errors
         for line in BufReader::new(file).lines() {
             if let Ok(entry) = line {
                 let mut parts = entry.split(' ');
 
                 let modified: SystemTime = {
-                    let str = parts.next().ok_or(ArklibError::Parse)?;
-                    UNIX_EPOCH.add(Duration::from_millis(
-                        str.parse().map_err(|_| ArklibError::Parse)?,
-                    ))
+                    match parts.next() {
+                        Some(millis) => match millis.parse() {
+                            Ok(millis) => {
+                                UNIX_EPOCH.add(Duration::from_millis(millis))
+                            }
+                            Err(_) => {
+                                log::warn!("Can't parse {millis} to u64");
```

Comment:
I think we should not continue the loop. If the file is broken, reading it makes no sense.